reference/src/layout/unions.md

```diff
@@ -8,12 +8,16 @@ not to change until an RFC ratifies them.
 
 ### Layout of individual union fields
 
-The main degree of freedom the compiler has when computing the layout of a union
-is to determine the offset of the fields. Union fields are laid out by the
-compiler "on top" of each other, that is, the bytes of one field might overlap
-with the bytes of another field - as opposed to, e.g., `struct`s, where the
-fields are laid out "next to" each other, such that the bytes of one field never
-overlap with the bytes of another field. This can be visualized as follows:
+A union consists of several variants, one for each field. These variants are
+laid out "on top" of each other, so they must all have the same size. That is,
+the byte of each variant overlaps with the byte located at the same offset in
+all other variants, and the bytes of one field might overlap with the bytes of
+another field - as opposed to, e.g., `struct`s, where the fields are laid out
+"next to" each other, such that the bytes of one field never overlap with the
+bytes of another field. The main degree of freedom the compiler has when
```

Comment:
Wow, this is the most convoluted explanation of "overlap" I have ever seen. Why do you deem this necessary?

Also, if we say "the main degree of freedom" that leaves the question "what are the others". Are there any others?

Reply:
I've summarized this further